doc/esp8266wifi/generic-class.rst

```diff
@@ -42,6 +42,33 @@ persistent
 
     WiFi.persistent(persistent)
 
+Starting from version 3 of this core, **persistence is disabled by default
+and WiFi does not anymore automatically fires up at boot** (see PR `#7902 <https://github.com/esp8266/Arduino/pull/7902>`__).
+
+Previously, SDK was automatically starting WiFi at boot.  This was probably
```

Comment:
I suggest highlighting here that the prior to v3 persistent was defaulted to true, and so the wifi creds were always stored to flash when the WiFi was configured, which could lead to a reduced life of the wifi config flash sector, and hence the entire ESP. Starting with v3, persistent defaults to false, so creds aren't stored by the SDK in the flash sector and hence don't survive across reboots.

Reply:
@devyte I think this is not precisely correct. Settings that were once persisted, remain that way, so thereafter, `WiFi.begin()`, with no `WiFi.persistent(true)` in the sketch, as long as `enableWiFiAtBootTime()` is somewhere in the sketch, will suffice to auto-connect to the AP.